integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -224,7 +224,11 @@ def test_all_null_int96(spark_tmp_path):
     class AllNullTimestampGen(TimestampGen):
         def start(self, rand):
             self._start(rand, lambda : None)
-    data_path = spark_tmp_path + '/PARQUET_DATA'
+
+        def _cache_repr(self):
```

Comment:
So we need to add a `_cache_repr` even if it will only be used for once to prevent a key conflict in cache. Didn't realized it, nice catch.